src/componentTypesView.ts

```diff
@@ -197,15 +198,16 @@ export class ComponentTypesView implements TreeDataProvider<ComponentType> {
 
         const regURL = await window.showInputBox({
             ignoreFocusOut: true,
-            prompt: 'Provide registry URL to display in the view',
+            value: registryContext?.URL,
+            prompt: registryContext ? 'Edit registry URL' : 'Provide registry URL to display in the view',
             placeHolder: 'Registry URL',
             validateInput: async (value) => {
                 const trimmedValue = value.trim();
                 if (!validator.isURL(trimmedValue)) {
                     return 'Entered URL is invalid';
                 }
                 const registries = await ComponentTypesView.instance.getRegistries();
-                if (registries.find((registry) => registry.URL === value)) {
+                if (!registryContext && registries.find((registry) => registry.URL === value)) {
```

Comment:
This disables validation for registry with the same URL to avoid validation error for registryContext initial URL and for other registries. The condition should be a part of find method callback instead.

Reply:
@dgolovin If user wants to edit the registry name alone than URL, it should allow the user to do. So I have made the condition based on **registryContext** value. It has value only at the time of **edit**, suppose if user wants to add new registry by that time **registryContext** value would be undefined and the other condition will check if the URL already exists/not.

Reply:
Let's do thought experiment. If you have two registries:
1. name: r1, url: url1
2. name: r2, url: url2
Now if you want to edit registry 1, you can call edit and change name to r3 and url to url2 and you wont see a validation error, because validation for url is going to be disabled.